blockchain/blockdao/blob_store.go

```diff
@@ -85,13 +89,17 @@ func (bs *blobStore) Start(ctx context.Context) error {
 	if err := bs.checkDB(); err != nil {
 		return err
 	}
-	if bs.interval != 0 {
-		bs.purgeTask = routine.NewRecurringTask(func() {
-			bs.expireBlob(atomic.LoadUint64(&bs.currWriteBlock))
-		}, bs.interval)
-		return bs.purgeTask.Start(ctx)
+	if bs.interval == 0 {
+		// default to 1 hour
+		bs.interval = time.Hour
 	}
-	return nil
+	bs.purgeTask = routine.NewRecurringTask(func() {
+		height, err := bs.tipHeightHandler()
+		if err != nil {
+			bs.expireBlob(height)
+		}
+	}, bs.interval)
```

Comment:
the timing of write to blob store could vary a lot, 
should use the DAO's tip height as baseline to expire/purge blobs, this way it is a consistent expire batch size = 720 blocks every time `expireBlob()` is called